app/controllers/users_controller.rb

```diff
@@ -8,14 +8,20 @@ def new
 
   def create    
     @user = User.new(user_params)
-    if @user.save
-      set_following_relationship
-      charge_user_with_stripe
-      AppMailer.delay.send_welcome_email(@user)
-      session[:user_id] = @user.id
-      redirect_to home_path
+    if @user.valid?    
+      charge = charge_user_with_stripe
+      if charge.success?
+        @user.save
+        set_following_relationship
```

Comment:
This looks like it is set on every registration but I think it is rather handling the invitation. I would consider giving it slightly different name. What do you think?